runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/DataUtil.java

```diff
@@ -23,20 +25,23 @@ private DataUtil() {
   /**
    * Serializes the elements in a non-serialized partition into an output stream.
    *
-   * @param coder                  the coder to encode the elements.
+   * @param serializer             the serializer to encode the elements.
    * @param nonSerializedPartition the non-serialized partition to serialize.
    * @param bytesOutputStream      the output stream to write.
    * @return total number of elements in the partition.
    * @throws IOException if fail to serialize.
    */
-  public static long serializePartition(final Coder coder,
+  public static long serializePartition(final Serializer serializer,
                                         final NonSerializedPartition nonSerializedPartition,
                                         final ByteArrayOutputStream bytesOutputStream) throws IOException {
     long elementsCount = 0;
+    final Coder coder = serializer.getCoder();
+    final OutputStream wrappedStream = buildOutputStream(bytesOutputStream, serializer.getChainables());
```

Comment:
Why don't we wrap stream in `convertToSerPartitions`?
Closing given `bytesOutputStream` with the wrapped streams seems not good.

Reply:
I wanted to show `coder.encode()` and `buildOutputStream()` should be called together.